### DISCUSSION TITLE
Z steps skipped on Ender 3V2

### DISCUSSION THREAD

**Discussion by Miq1** [Jyers/Marlin#1754] [Q&A]
Hi guys,
I have been using the prebuilt 2.0.1 bugfix version with UBL without sensor and 5x5 mesh for some weeks on my Ender 3V2 with 4.2.2 mainboard without any problem.
Since a couple of days I am experiencing Z steps being skipped, leading to nearly all prints being for the trash only.
This may be an issue of the stepper driver or stepper motor (I sorted out all mechanical causes I could think of), but I wanted to sort out the firmware as well, if possible.
So: are there any reports or experiences for the firmware to skip Z steps at all? A "naught" would be appreciated as well 😉

**Comment by Cal-LF** [comment]
I have not experienced any Z skips and not read of any. Possible the board is failing.

**Comment by Miq1** [comment]
Thanks guys!

That was what I had anticipated in fact.

I have the mainboard fan connected to the Hotend cooler terminals, Vref is at about 1.21V for all X, Y and Z (Extruder is at 1.39). The motor coupler is firmly fixed, the threaded rod is rotatable without much resistance at any height. X axis rollers are tight, but turning normally. X and Y dimensions of printed objects are flawless.

I attached a scale to the upper end of the threaded rod and moved the Z stepper several times up and down. After that the scale was off by at least 5 degrees, and repeatable so, so I think now I have narrowed it down to the stepper driver or stepper motor as root cause.

I contacted Creality support with this info - we will see if they take responsibility...

**Reply by Cal-LF** [reply]
@Miq1 Vref is too high in my opinion. I have all mine set to 0.9 to 1.0

**Reply by Miq1** [reply]
Really? I read of 1.22V being ideal, so I did not worry about mine. The lower the voltage the higher the current I was told. Would not the steppers get much hotter?

**Reply by Cal-LF** [reply]
@Miq1 The higher the voltage the higher the current. The steppers will operate cooler with reduced Vref. 1.21v for the stepper drivers is a recommended setting that I have also read. If the Z stepper is skipping at that setting, it may skip more with a lower setting.
I have had 2 E3V2 with defective main boards skipping all the steppers. A bad batch of boards caused this. As a precaution with a replacement V427 board, I put bigger fans in place of the stock fans. 
This issue is known for some E3V2 printers from November 2020 time frame. If cooling the mainboard does not fix the skipping, a replacement may be needed.

**Reply by Miq1** [reply]
I see. I do not know when my printer was manufactured, I got it in September 2021. It still has a 4.2.2 board, though. I have replaced the stock fan by a Sunon one, but again as a 4010 type. I did not want to modify too much for a beginning, but that will change gradually. 😆

**Reply by Miq1** [reply]
That slightly is confusing me. I observed a full rotation more or less equalling 8mm, so 5 degrees is 8/360*5=1/9mm. And keep in mind the measured difference is only what remains from missed steps up and down, where a skip upwards is compensated by another skip on the way down.

**Reply by Miq1** [reply]
I see where my irritation is coming from: I was talking stepper steps, whereas you were referring to micro steps. So my calculation looks correct.
